cabby/geo/geo_item.py

```diff
@@ -14,67 +14,126 @@
 '''Basic classes and functions for RVSPath items.'''
 
 import geopandas as gpd
-from geopandas import GeoDataFrame, GeoSeries
 import pandas as pd
-import re
-from shapely.geometry.point import Point
-from shapely.geometry import box, mapping, LineString
-from typing import Text, Dict
+from typing import Any, Dict, Sequence
+import os
+from shapely.geometry import LineString, Polygon
+
+from cabby.geo import util
 
 import attr
 
+_Geo_DataFrame_Driver = "GPKG"
+VERSION = 0.4
 
 @attr.s
-class GeoPath:
-  """Construct a RVSPath sample.
-  `start_point` is the beginning location.
-  `end_point` is the goal location.
-  `path_features` is a path features: cardinal directions, intersections, instructions and route geometry.
-  `main_pivot` is pivot along the route.
-  `near_pivot` is the pivot near the goal location.
-  `beyond_pivot` is the pivot beyond the goal location.
+class GeoEntity:
+  """Construct a geo entity.
+
+  `geo_landmarks` the geo landmarks (start and end points + pivots).
+  `geo_features` the spatial features of the path.
+  `route` the path from the start to end point.
   """
-  start_point: GeoDataFrame = attr.ib()
-  end_point: GeoDataFrame = attr.ib()
-  path_features: GeoDataFrame = attr.ib()
-  main_pivot: GeoDataFrame = attr.ib()
-  near_pivot: GeoDataFrame = attr.ib()
-  beyond_pivot: GeoDataFrame = attr.ib()
-  size: int = attr.ib(init=False)
+
+  geo_landmarks: Dict = attr.ib()
+  geo_features: Dict = attr.ib()
+  route: gpd.GeoDataFrame = attr.ib()
+
+  @classmethod
+  def add_entity(cls,
+                 route: gpd.GeoDataFrame,
+                 geo_landmarks: Dict[str, gpd.GeoDataFrame],
+                 geo_features: Dict[str, Any]):
+    geo_entity = GeoEntity({}, geo_features, route)
+    for landmark_type, landmark in geo_landmarks.items():
+      geo_landmark = GeoLandmark.add_pivot(landmark, landmark_type)
+      geo_entity.geo_landmarks[geo_landmark.landmark_type] = geo_landmark
+    return geo_entity
+
+  @staticmethod
```

Comment:
@staticmethod in classes doesn't conform to the style guide: https://google.github.io/styleguide/pyguide.html#2174-decision

Please pull this out as a module level function.